solr/core/src/java/org/apache/solr/search/stats/ExactStatsCache.java

```diff
@@ -230,10 +230,10 @@ protected void doSendGlobalStats(ResponseBuilder rb, ShardRequest outgoing) {
       Map<String,TermStats> globalTermStats = new HashMap<>();
       Map<String,CollectionStats> globalColStats = new HashMap<>();
       // aggregate collection stats, only for the field in terms
-      String collectionName = rb.req.getCore().getCoreDescriptor().getCollectionName();
-      if (collectionName == null) {
+      String collectionName = rb.req.getCloudDescriptor().getCollectionName();
+     /* if (collectionName == null) {
         collectionName = rb.req.getCore().getCoreDescriptor().getName();
-      }
+      }*/
```

Comment:
Why can't this be null anymore? It was a strange check to begin with, makes me assume there was a subtle edge case here.